app/models/employee.rb

```diff
@@ -19,19 +19,36 @@ class Employee < ApplicationRecord
   belongs_to :company_branch
   has_many :attendances, dependent: :destroy
 
+  validates :name, :email, :position, :employee_number, :private_number, presence: true
+  validates :email, uniqueness: true
+  validates :private_number, uniqueness: true
+  validates :employee_number, uniqueness: true
+
   scope :actives, -> { where(is_active: true) }
 
+  def self.search(search_name)
+    if search_name
+      self.where('name LIKE ?', "%#{search_name}%")
+    else
+      self.order(:name)
+    end
+  end
```

Comment:
I suggest to add specs for this, in the same way you did for CompanyBranch model, and in the same way for methods absence_by_month, and average. That would give clarity to the code in general.

Reply:
Yep we will write it. Thanks